app/main.py

```diff
@@ -1 +1,17 @@
-# write your code here
+from app.errors import VaccineError, NotWearingMaskError
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+
+    masks_to_buy = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
```

Comment:
Catching the general `VaccineError` here will also catch `OutdatedVaccineError`, which might not be the intended behavior if you want to differentiate between not vaccinated and outdated vaccine cases. Consider catching `NotVaccinatedError` and `OutdatedVaccineError` separately if different handling is needed.